DependencyInjection/Compiler/LoggerChannelPass.php

```diff
@@ -81,7 +81,7 @@ public function process(ContainerBuilder $container)
 
         // create additional channels
         foreach ($container->getParameter('monolog.additional_channels') as $chan) {
-            $loggerId = sprintf('monolog.logger.%s', $chan);
+            $loggerId = $chan === 'app' ? 'monolog.logger' : sprintf('monolog.logger.%s', $chan);
```

Comment:
this pass should not create the `monolog.logger` service, as it already exists (and the `logger` alias is already public, even though the `monolog.logger` service is not, and was never public)

Reply:
a better fix would be to skip `app` entirely in this loop, as it is special.

Reply:
If someone declares `app` as a `monolog:channel`, don't you think they'd expect the `monolog.logger` to be `public`?

Reply:
we never made `monolog.logger` public, even before the Symfony change to make services private by default. The public name of the main logger has always been `logger`

Reply:
And oh, by the way: I just got 

> The "logger" service is private, getting it from the container is deprecated since Symfony 3.2 and will fail in 4.0. You should either make the service public, or stop using the container directly and use dependency injection instead.

... in a logfile. So, what would be the right approach to make `logger` public?

Reply:
@stof *see above*